### PR TITLE
Use UidNumber and GidNumber fields in User objects

### PR DESCRIPTION
**Pull request by sudo-sturbia**

### CODE REVIEW

File: pkg/cbox/user/rest/rest.go [hunk 1]
```diff
@@ -507,12 +497,8 @@ func (m *manager) IsInGroup(ctx context.Context, uid *userpb.UserId, group strin
...
-		if uidObj, ok := u.Opaque.Map["uid"]; ok {
-			if uidObj.Decoder == "plain" {
-				return string(uidObj.Value), nil
-			}
-		}
+	if u.UidNumber == 0 {
+		return "", errors.New("rest: could not retrieve UID from user")
 	}
-	return "", errors.New("rest: could not retrieve UID from user")
+	return fmt.Sprintf("%v", u.UidNumber), nil
```
Comment (sudo-sturbia) [inline pkg/cbox/user/rest/rest.go:503]: There are several parts of code similar to this, where the Opaque map was checked before returning a value. I chose to check if UidNumber and GidNumber are zeros. This only works if uid and gid can't be zeros, so I'd appreciate if someone can verify this.

Comment (ishank011) [reply pkg/cbox/user/rest/rest.go:503]: The (0, 0) values are reserved for root users, so we can assume that none of the ordinary users would have these. That being said, in cases where we're unable to retrieve the UID or GID, these might be treated as privileged users and granted access to resources they're not supposed to access. Can you find such instances in eosfs and add checks? If a user has these values, raise an error.

File: pkg/auth/manager/ldap/ldap.go [hunk 2]
```diff
@@ -182,7 +183,14 @@ func (am *mgr) Authenticate(ctx context.Context, clientID, clientSecret string)
 	if getGroupsResp.Status.Code != rpc.Code_CODE_OK {
 		return nil, errors.Wrap(err, "ldap: grpc getting user groups failed")
 	}
-
+	gidNumber, err := strconv.ParseInt(sr.Entries[0].GetEqualFoldAttributeValue(am.c.Schema.GIDNumber), 10, 0)
```
Comment (ishank011) [inline pkg/auth/manager/ldap/ldap.go:186]: Bit size 0? Should be 64

File: pkg/auth/manager/ldap/ldap.go [hunk 3]
```diff
@@ -191,17 +199,10 @@ func (am *mgr) Authenticate(ctx context.Context, clientID, clientSecret string)
 		Groups:      getGroupsResp.Groups,
 		Mail:        sr.Entries[0].GetEqualFoldAttributeValue(am.c.Schema.Mail),
 		DisplayName: sr.Entries[0].GetEqualFoldAttributeValue(am.c.Schema.DisplayName),
+		UidNumber:   uidNumber,
+		GidNumber:   gidNumber,
 		Opaque: &types.Opaque{
```
Comment (ishank011) [inline pkg/auth/manager/ldap/ldap.go:204]: Just don't initialize opaque here. It should be nil. Ditto in all the instances.

File: pkg/storage/utils/eosfs/eosfs.go [hunk 4]
```diff
@@ -1457,23 +1457,13 @@ func getResourceType(isDir bool) provider.ResourceType {
...
-			if gidObj.Decoder == "plain" {
-				gid = string(gidObj.Value)
-			}
-		}
+	if u.UidNumber == 0 {
+		return "", "", errors.New("eos: uid missing for user")
 	}
-	if uid == "" || gid == "" {
-		return "", "", errors.New("eos: uid or gid missing for user")
+	if u.GidNumber == 0 {
```
Comment (ishank011) [inline pkg/storage/utils/eosfs/eosfs.go:1463]: We need to make sure that UID = 0 or GID = 0 in user requests is blocked eosfs

File: pkg/auth/manager/ldap/ldap.go (hunk 2 above)
Comment (sudo-sturbia) [reply pkg/auth/manager/ldap/ldap.go]: Done

File: pkg/auth/manager/ldap/ldap.go (hunk 3 above)
Comment (sudo-sturbia) [reply pkg/auth/manager/ldap/ldap.go]: Done

File: pkg/cbox/user/rest/rest.go (hunk 1 above)
Comment (sudo-sturbia) [reply pkg/cbox/user/rest/rest.go:497]: It seems that functions in `eosfs` use `getUser` function for any user related operations, so i updated it to check for uid and gid in fabd746. Please let me know if that's what you had in mind.

File: pkg/storage/utils/eosfs/eosfs.go (hunk 4 above)
Comment (sudo-sturbia) [reply pkg/storage/utils/eosfs/eosfs.go:1469]: I'm not sure i understand, if you mean that the statement should be `if u.UidNumber == 0 || u.GidNumber == 0`, then you will notice that i separated it into two if statements to provide a more specific error message.

**Comment by ishank011** [comment]
@sudo-sturbia majorly looks good, nice work! I'll test it since you don't have access to an EOS instance so there might be a delay in merging. We're working on getting EOS tests integrated in the CI so we should have that soon

Meanwhile, please rebase and fix the failing integration tests.

**Comment by sudo-sturbia** [comment]
@ishank011 Hi, I thought I'd let you know the latest updates: I looked at the integration tests and was able to find the errors, the changes are mainly in 999730a and e685753. As for the old commits, struct tags for `UIDNumber` and `GIDNumber` in `pkg/auth/manager/json` were updated to match `User` struct. Currently all tests pass, except for `LGTM: JavaScript`, which is odd as i haven't updated any JavaScript, so I'd appreciate it if you let me know how to recreate it locally so i can maybe find the problem.

**Review by ishank011 — APPROVED** [review]
@sudo-sturbia LGTM! Good work on finding the issues with the integration tests. The LGTM tool is not that mature yet, so we do see intermittent build failures sometimes; we can ignore that. I'll test the changes tomorrow and merge.

**Comment by sudo-sturbia** [comment]
> @sudo-sturbia LGTM! Good work on finding the issues with the integration tests. The LGTM tool is not that mature yet, so we do see intermittent build failures sometimes; we can ignore that. I'll test the changes tomorrow and merge.

That sounds great. Thank you!

**Review by ishank011 — APPROVED** [review]

**Pull request merged**